robot/impl/local_robot.go

```diff
@@ -1231,6 +1231,9 @@ func (r *localRobot) reconfigure(ctx context.Context, newConfig *config.Config,
 	}
 
 	if diff.ResourcesEqual {
+		// Even if resources are "equal," still update the value of `r.initializing`. We may
+		// be moving from a minimal config to an empty, fully processed config.
+		r.initializing.Store(newConfig.Initial)
```

Comment:
should we just do a defer here? since we always want to update `r.initializing`

Reply:
Yep. I couldn't see any `reconfigure` return paths in which we should _not_ be setting it. Thanks for calling that out.